docker/Dockerfile

```diff
@@ -10,64 +10,48 @@ ENV TERM linux
 ENV BASH_ENV ~/.bashrc
 SHELL ["/bin/bash", "-c"]
 
-# comment out the line in the default .bashrc that does an early exit if not in
-# interactive mode. It should be noted that the -i option of docker run does
-# not launch the shell in interactive mode. CCheck Dockerfile.3 for a demo.
-RUN sed -e '/[\ -z \"\$PS1\"\ ]\ &&\ return/ s/^#*/#/' -i /root/.bashrc
-
 # General installs in the docker
-RUN apt-get -y update && apt-get -y upgrade
-
-RUN apt-get -y install \
-    apt-utils \
-    build-essential \
-    software-properties-common
-
-RUN apt-get install -y \
-    curl \
-	git \
-	screen \
-	ufw \
-    vim \
-	wget
-
-# Install miniconda and add to PATH
-RUN wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
-RUN sh Miniconda3-latest-Linux-x86_64.sh -b
-ENV PATH /root/miniconda3/bin:$PATH
-
-# expose ports for flask app - both server and backoffice
-EXPOSE 8888 8890
+RUN apt-get -y update \
+    && apt-get install -y git \
+       wget \
+    && rm -rf /var/lib/apt/lists/*
 
 # Create app directory and copy code
 RUN mkdir /home/icubam
 WORKDIR /home/icubam
 
-# must copy explicitely to handle cases where image is build in an environment without extra files (db, config)
-# or with extra file (e.g., after a first init).
+# must copy explicitely to handle cases where image is build in an environment with and without extra files
+# (db, config) in the context (e.g., relaunching after a first init).
+
 # TODO:  download code directly in the container from repository/package instead orf relying on local files
 
 # get docker config and move startup scripts for the server and sms containers
 COPY docker docker
-RUN mv ./docker/start_server.sh .
-RUN mv ./docker/start_server_sms.sh .
-RUN mv ./docker/start_server_bo.sh .
 # copy code and scripts
 COPY icubam icubam
 COPY scripts scripts
+
 # create resource folder
 RUN  mkdir resources
+
 # copy specific files on root folder
-COPY doc/install.md LICENSE pyproject.toml README.md requirements.txt setup.py ./
+COPY LICENSE pyproject.toml README.md requirements.txt setup.py ./
+
 # pip/setup tools requires version inside .git
 COPY .git ./.git
 
-# prepare for the conda environment
-RUN echo ".  /root/miniconda3/etc/profile.d/conda.sh" >> ~/.bashrc
-RUN conda create -n icubam python=3.8
-RUN conda activate icubam && pip install -r requirements.txt && pip install -e .
+RUN pip install -r requirements.txt && pip install -e .
+
+# clear pip cache
+RUN rm -rf ~/.cache/pip
+
+# make sure final image is clean
+FROM builder as launcher
```

Comment:
What does this do? Just curious.

Reply:
If we don't do multi-stage (here having the "builder" intermediate layer) then the `rm` removes the files for the containers that will be started, but in terms of docker layers (hence image size) this has no effect (the layers where the files have been added is still in the "history" of layers). Building an intermediate image allow to solve this.